src/DetectChanges/BCBreak/FunctionBased/ParameterTypeContravarianceChanged.php

```diff
@@ -70,7 +78,18 @@ private function typeToString(?ReflectionType $type): string
             return 'no type';
         }
 
-        return ($type->allowsNull() ? '?' : '')
-            . $type->__toString();
+        return $type->__toString();
+    }
+
+
+    /** @TODO may the gods of BC compliance be merciful on me */
```

Comment:
May the gods have mercy of my.

This is ugly, but it highlights a design limitation when trying to traverse a whole reflection graph.

Perhaps @kukulich has a better idea on how to achieve this, but the problem is basically that we're trying to find `parent` classes and related interfaces by their name, and it is not possible to do so from a `ReflectionType` from `roave/better-reflection:5.0.0`, as `ReflectionType` is mostly immutable.

Instead, we have to fetch the associated reflector here.

Injecting the reflector upfront is not feasible either here, with the current design (the reflector is built at runtime, by some services that do run a `composer install`) :thinking:

Reply:
https://github.com/Roave/BetterReflection/pull/836